apps/io-func-wallet-solution-backend/src/entity-configuration.ts

```diff
@@ -0,0 +1,114 @@
+import * as t from "io-ts";
+import * as TE from "fp-ts/TaskEither";
+import * as E from "fp-ts/Either";
+import * as RTE from "fp-ts/ReaderTaskEither";
+import { pipe } from "fp-ts/function";
+import { sequenceS } from "fp-ts/lib/Apply";
+
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { UrlFromString } from "@pagopa/ts-commons/lib/url";
+
+import { validate } from "./validation";
+import {
+  GRANT_TYPE_KEY_ATTESTATION,
+  LoA,
+  RELATIVE_TOKEN_ENDPOINT,
+  TOKEN_ENDPOINT_AUTH_METHOD_SUPPORTED,
+  getLoAUri,
+} from "./wallet-provider";
+import { JwkPublicKey } from "./jwk";
+import { Signer } from "./signer";
+import { EntityConfigurationToJwtModel } from "./encoders/entity-configuration";
+
+export const FederationEntityMetadata = t.type({
+  basePath: UrlFromString,
+  organizationName: NonEmptyString,
+  homePageUri: UrlFromString,
+  policyUri: UrlFromString,
+  tosUri: UrlFromString,
+  logoUri: UrlFromString,
+});
+
+export type FederationEntityMetadata = t.TypeOf<
+  typeof FederationEntityMetadata
+>;
+
+type EntityConfigurationEnvironment = {
+  federationEntityMetadata: FederationEntityMetadata;
+  signer: Signer;
+};
+
+const FederationEntity = t.type({
+  organizationName: t.string,
+  homepageUri: t.string,
+  policyUri: t.string,
+  tosUri: t.string,
+  logoUri: t.string,
+});
+
+const WalletProviderMetadataPayload = t.type({
+  jwks: t.array(JwkPublicKey),
+  tokenEndpoint: t.string,
+  tokenEndpointAuthSigningAlgValuesSupported: t.array(t.string),
+  ascValues: t.array(t.string),
+  grantTypesSupported: t.array(t.string),
+  tokenEndpointAuthMethodsSupported: t.array(t.string),
+});
+
+export const EntityConfigurationPayload = t.type({
+  iss: t.string,
+  sub: t.string,
+  walletProviderMetadata: WalletProviderMetadataPayload,
+  federationEntity: FederationEntity,
+});
+
+export type EntityConfigurationPayload = t.TypeOf<
+  typeof EntityConfigurationPayload
+>;
+
+export const getEntityConfiguration =
+  (): RTE.ReaderTaskEither<EntityConfigurationEnvironment, Error, string> =>
```

Comment:
*suggestion(readability)*: as it's a long pipe, consider use comments to explain relevant steps. Purpose: to help readers understand what the pipe does just by reading, without the need to "mentally execute" the whole code

Reply:
> Maybe a comment could be added about the whole feature, what do you think?

Yes, that would be great. The point is I don't need to understand the implemented algorithm to know what the function does.